source/normalize-arguments.ts

```diff
@@ -94,23 +105,23 @@ export const preNormalizeArguments = (options: any, defaults?: any) => {
 	}
 
 	if (options.dnsCache) {
-		const cacheableLookup = new CacheableLookup({cacheAdapter: options.dnsCache});
-		options.lookup = cacheableLookup.lookup;
+		const cacheableLookup = new CacheableLookup({cacheAdapter: options.dnsCache as Keyv | undefined});
```

Comment:
I'm not a TypeScript expert, so this may sound dumb. Is `Keyv | undefined` really needed? CacheableLookup has already defined `cacheAdapter?: Keyv;`

Reply:
The type for cacheAdapter can also be a Map, due to the usage I saw... Unless that's incorrect, in which case I can correct the type and remove the cast

Reply:
You're right. I need to update the types then.